distribution/src/agent.js

```diff
@@ -75,75 +101,97 @@ function provideInitialize(data) {
     ];
 
     const decimalsAbi = ['function decimals() view returns (uint8)'];
-    const compAddress = compToken.address.toLowerCase();
-    const compContract = new ethers.Contract(compAddress, decimalsAbi, data.provider);
-    const compDecimals = await compContract.decimals();
+    const compoundTokenAddress = compoundToken.address.toLowerCase();
+    const compoundTokenContract = new ethers.Contract(
+      compoundTokenAddress, decimalsAbi, data.provider,
+    );
+    const compoundTokenDecimals = await compoundTokenContract.decimals();
 
-    data.compDecimalsMultiplier = new BigNumber(10).pow(compDecimals);
-    data.compAddress = compAddress;
-  }
+    data.compoundTokenDecimalsMultiplier = new BigNumber(10).pow(compoundTokenDecimals);
```

Comment:
This operation is mixing `ethers.js` and `bignumber.js` `bignumbers` since the value of `compoundTokenDecimals` came from a contract call. Converting `compoundTokenDecimals` to a string should prevent any inconsistent behavior:
```suggestion
    data.compoundTokenDecimalsMultiplier = new BigNumber(10).pow(compoundTokenDecimals.toString());
```